pkg/e2e/operators/ocmagent.go

```diff
@@ -2,218 +2,225 @@ package operators
-				}
-				return true, nil
-			})
-			Expect(err).NotTo(HaveOccurred())
-		}, pollingDuration.Seconds())
-
-		util.GinkgoIt("ocm-agent service should be restored when it gets deleted", func(ctx context.Context) {
-			err := deleteService(ctx, ocmAgentServiceName, operatorNamespace, h)
-			Expect(err).NotTo(HaveOccurred())
-
-			err = wait.Poll(30*time.Second, 5*time.Minute, func() (bool, error) {
-				obj, err := h.Dynamic().Resource(schema.GroupVersionResource{
-					Group: "", Version: "v1", Resource: "services",
-				}).Namespace(operatorNamespace).Get(ctx, ocmAgentServiceName, metav1.GetOptions{})
-				if err != nil {
-					return false, fmt.Errorf("unable to retrieve ocm-agent service")
-				}
+	ginkgo.It("cluster role bindings exist", label.Install, func(ctx context.Context) {
+		clusterRoleBindings, err := h.Kube().RbacV1().ClusterRoleBindings().List(ctx, metav1.ListOptions{})
+		expect.NoError(err)
+		found := false
+		for _, clusterRoleBinding := range clusterRoleBindings.Items {
+			if strings.HasPrefix(clusterRoleBinding.Name, clusterRolePrefix) {
+				found = true
+			}
+		}
+		Expect(found).To(BeTrue())
+	})
 
-				if obj.GetName() != ocmAgentServiceName {
-					return false, fmt.Errorf("ocm-agent service not found")
-				}
-				return true, nil
-			})
-			Expect(err).NotTo(HaveOccurred())
-		}, pollingDuration.Seconds())
-
-		util.GinkgoIt("ocm-agent-allow-only-alertmanager  networkpolicy should restored when it gets deleted", func(ctx context.Context) {
-			err := deleteNetworkPolicy(ctx, ocmAgentNetworkpolicyName, operatorNamespace, h)
-			Expect(err).NotTo(HaveOccurred())
-
-			err = wait.Poll(30*time.Second, 5*time.Minute, func() (bool, error) {
-				obj, err := h.Dynamic().Resource(schema.GroupVersionResource{
-					Group: "networking.k8s.io", Version: "v1", Resource: "networkpolicies",
-				}).Namespace(operatorNamespace).Get(ctx, ocmAgentNetworkpolicyName, metav1.GetOptions{})
-				if err != nil {
-					return false, fmt.Errorf("unable to retrieve ocm-agent-allow-only-alertmanager  networkpolicy")
-				}
+	ginkgo.It("can be upgraded from previous version", label.Upgrade, func(ctx context.Context) {
+		performUpgrade(ctx, h, namespace, operatorName, operatorName, operatorRegistry, 5, 30)
+	})
 
-				if obj.GetName() != ocmAgentNetworkpolicyName {
-					return false, fmt.Errorf("ocm-agent-allow-only-alertmanager  networkpolicy not found")
-				}
-				return true, nil
-			})
-			Expect(err).NotTo(HaveOccurred())
-		}, pollingDuration.Seconds())
+	ginkgo.It("deployment is restored when removed", func(ctx context.Context) {
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deploymentName,
+				Namespace: namespace,
+			},
+		}
+
+		err := client.Delete(ctx, deployment)
+		expect.NoError(err)
+
+		err = wait.For(conditions.New(client).ResourceMatch(
+			deployment, func(object k8s.Object) bool {
+				obj := object.(*appsv1.Deployment)
+				return obj.Status.AvailableReplicas >= 1
```

Comment:
This wait can be simplified using one of the provided conditions like this:

```
wait.For(conditions.New(client).DeploymentConditionMatch(deployment, appsv1.DeploymentAvailable, v1.ConditionTrue))
```

Reply:
Done!